pkg/network/ebpf/c/protocols/http2/decoding.h

```diff
@@ -581,14 +589,15 @@ static __always_inline void find_relevant_frames(struct __sk_buff *skb, skb_info
     }
 
     // Checking we can read HTTP2_FRAME_HEADER_SIZE from the skb - if we can, update telemetry to indicate we have
-    if ((iteration == HTTP2_MAX_FRAMES_TO_FILTER) && (skb_info->data_off + HTTP2_FRAME_HEADER_SIZE <= skb_info->data_end)) {
-        __sync_fetch_and_add(&http2_tel->exceeding_max_frames_to_filter, 1);
-    }
+    bool more_frames_to_filter = ((iteration == HTTP2_MAX_FRAMES_TO_FILTER) && (skb_info->data_off + HTTP2_FRAME_HEADER_SIZE <= skb_info->data_end));
```

Comment:
1. The declaration should be closer to the usage (a.k.a line 600)
2. The value is being used only for the return, but the return statement have another condition. I think we should simply merge both (a.k.a, `return (iteration == HTTP2_MAX_FRAMES_TO_FILTER) && (skb_info->data_off + HTTP2_FRAME_HEADER_SIZE <= skb_info->data_end) && iteration_value->frames_count < HTTP2_MAX_FRAMES_ITERATIONS`)